Automattic/jetpack
# Newsletter: Retrieve current user email

Fixes https://github.com/Automattic/jetpack/issues/39391
Requires D163040-code

## Proposed changes:

Jetpack Subscribe blocks inserted on WordPress.com Atomic sites cannot know if the user is logged in into `wordpress.com` because the auth cookies are in a different domain (see https://github.com/Automattic/wp-calypso/issues/52046).

But they ultimately use the email of the user logged in into wp.com because the subscription happens on the `subscribe.wordpress.com` iframe, ignoring any email entered by the user on the email input:

Email input | Subscription modal
--- | ---
![image](https://github.com/user-attachments/assets/2f7147bc-284d-480b-9103-1866254568f1) | ![image](https://github.com/user-attachments/assets/d2de1ecf-4d01-4ae7-b834-e74ad0dffb88)

[We do try to disable the email input if the user is logged in](https://github.com/Automattic/jetpack/blob/27be71338e8b13719fc7fcf1b340bdaf35d3a2e6/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L829), but this doesn't work because of the issue mentioned above.

So, as a workaround, this PR communicates with the `subscriber.wordpress.com` iframe to determine the email of the current user.

Before | After
--- | ---
<img width="792" alt="Screenshot 2024-10-03 at 15 36 11" src="https://github.com/user-attachments/assets/78cbf40e-d35e-4ef5-aa0d-9f62dc1ada48"> | <img width="802" alt="Screenshot 2024-10-03 at 15 37 02" src="https://github.com/user-attachments/assets/77c256b4-cfc4-4b89-87b8-6a76c98fa6d7">


### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply D163040-code to your sandbox
- Sandbox `subscribe.wordpress.com`
- Apply this PR to a WoA site
- Publish a post with a Jetpack Subscribe block
- In a different browser session, go to `wordpress.com` and log in with a different user
- Visit the published post
- Make sure the email input has your email and has been disabled
